Add support for Quectel LC29H-BS

deece opened the pull request:
Please review.

This patch set adds preliminary support for the Quectel LC29H-BS.

A new tool tools/nmea.py is introduced to allow NMEA commands to be sent via a serial port. This is used to detect the device, as well as on-boot configuration.

The settings requesting NMEA messages do not seem to be saved in flash, so we introduce a systemd unit to configure the GPS unit before str2str_tcp starts.

Test request: Please test that the install script still works in the original configuration.

Stefal commented:
Hi!

Thank you!
I will get a Quectel LG290P in a few weeks, and will start to work on it after I'll have added the UM980 support.
I hope that the config methods are similar between these two receivers.

longcat99 commented:
![image](https://github.com/user-attachments/assets/9fea557d-0272-4b93-bffb-964343d2c64e)
I use this pr, which is great, but it cannot be installed via install.sh. The py program will read some other instructions. In addition, I translated the web into Chinese. If I need to submit a PR.
![image](https://github.com/user-attachments/assets/dd6e9350-f477-4910-b0f8-d11c19e5b926)

MinimumDowntime commented:
I have @deece's branch sort of running, but it took some effort (I probably made this way harder than it needed to be):

1. I installed Stefal's version (v2.6.3)
2. I copied the files down from deece's branch into the respective folders
3. I manually setup and configured the service (configure_gps.service) used in deece's branch, then rebooted.

Once back up, I had satellite signals (it took a minute or so before they appeared) and could output correction data. But there are a couple of things I can't get working:
- The map still shows the marker at the default location as though no coordinates have been received for the Base Station. That means the section for Latitude, Longitude, and Height all read 0's
- I saw this pull request had code that could detect/configure for the LC29H-BS, but those don't seem to work in the GUI for me. The device is never found via the UI, but again - the device _does_ work and outputs correction data as I can only assume the configure_gps.service has correctly handled all of this at boot

I'm guessing that I've missed something in my manual installation of this, the code doesn't understand this device somewhere along the way, or a mix of both. In any case, these are incredibly minor inconveniences as all still appears to work as expected for sending correction data over NTRIP.

deece commented:
The map doesn't update by default as the module needs to be configured on each boot of the device to output location data. 

If you run configure-gps.sh at boot, it will send the required commands for the module to output the right information.

deece committed:
Add support for Quectel LC29H-BS

Signed-off-by: Alastair D'Silva <alastair@d-silva.org>

deece committed:
A helper tool for surveying in the LC29HBS RTK GPS

It initiates survey-in on the device, and also calculates and displays the average geodetic
coordinates to manually set the base station location in RTKBase

Example run:
python3 lc29h-bs_survey.py --mode survey --min-dur 86400 --speed 921600 /dev/ttyS0

erok26 commented:
@deece,
Could you update your install.sh and your configure_gps.sh so that they use the name "Quectel_LC29HBS" as putting the Underscore in the device model/type would then allow it on the WebUI instead of having to update the settings.conf manually. I believe these are the only 2 places that you are calling the model/type name.
Regardless, even after I run configure_gps.sh, I still don't get $GGA nmea sentences. It still comes in via binary which makes me wonder if I'm on an older version of firmware.

codeCraft-Ritik reviewed (commented):
Nice addition! Supporting the LC29H-BS while keeping the existing workflow intact is a valuable enhancement.
The introduction of tools/nmea.py also provides a clean foundation for future GPS device integrations 👍
